java-checks/src/main/java/org/sonar/java/checks/SwitchWithTooManyCasesCheck.java

```diff
@@ -19,23 +19,21 @@
  */
 package org.sonar.java.checks;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import org.sonar.check.Rule;
 import org.sonar.check.RuleProperty;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
 import org.sonar.plugins.java.api.semantic.Type;
 import org.sonar.plugins.java.api.tree.CaseGroupTree;
-import org.sonar.plugins.java.api.tree.SwitchStatementTree;
+import org.sonar.plugins.java.api.tree.SwitchTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-
 @Rule(key = "S1479")
 public class SwitchWithTooManyCasesCheck extends IssuableSubscriptionVisitor {
```

Comment:
I've just realized I have some doubts about exactly this rule and a Switch Expression. As far as I understand in the expression you need to cover all the cases, otherwise it's a compile-time error. So if we have an Enum with 10 values or a sealed class in the future. It could bring some FPs, because you have to cover all cases. wdyt?

Reply:
In fact, that already the case for Switch Statements. You don't have to, but covering all possible values of an enum is something you might want to do. This is why this rule is not triggered with enum. it might become an issue for sealed classes, we might want to exclude them in the future.